src/main/java/at/hannibal2/skyhanni/features/misc/IslandLeaveJoinMsgs.kt

```diff
@@ -0,0 +1,144 @@
+        val islandOwners = mutableListOf<String>()
+
+        val joinMessage = " §" + if (config.leaveJoinColor) { "a" } else { "e" } + "joined §ethe island."
+        val leaveMessage = " §" + if (config.leaveJoinColor) { "c" } else { "e" } + "left §ethe island."
+
+        for (line in event.tabList) {
+            if (guesting && !onKnownIsland) {
+                islandCategoryPattern.matchMatcher(line) {
+                    inIslandCategory = true
+                }
+                if (inIslandCategory) offlinePlayerPattern.matchMatcher(line) {
+                    islandOwners.add(line)
+                }
+                guestCategoryPattern.matchMatcher(line) {
+                    inIslandCategory = false
+                    for (player in islandOwners) {
+                        if (isPlayerKnown(player.removeColor())) {
+                            onKnownIsland = true
+                            return@matchMatcher
+                        }
+                    }
+                    return@matchMatcher
+                }
+            }
+
+            cleanPlayerPattern.matchMatcher(line) {
+                val player = group("player")
+                cleanPlayersNew.add(player)
+                if (guesting && inIslandCategory) {islandOwners.add(player)}
+                if (players.contains(player)) {
+                    return@matchMatcher
+                }
+                players.add(player)            // !onPrivateIslandGarden because a vanilla message gets sent
+                if (shouldSendMsg(player) && updatedSinceWorldSwitch && !onPrivateIslandGarden) {
+                    ChatUtils.chat(player + joinMessage)
+                }
+            }
+            rawPlayerPattern.matchMatcher(line) {
+                rawPlayersNew.add(group("player"))
+            }
+        }
+
+        if (players.size > 1) {
+            updatedSinceWorldSwitch = true
+        }
+
+        if (players.isEmpty()) return
+
+        for ((index, player) in players.withIndex().reversed()) {
+            if (!rawPlayersNew.contains(player) && !cleanPlayersNew.contains(player)) {
+                if (shouldSendMsg(player)) ChatUtils.chat(player + leaveMessage)
+                players.removeAt(index)
+            }
+        }
+    }
+
+    @SubscribeEvent
+    fun onWorldChange(event: LorenzWorldChangeEvent) {
+        clearPlayers()
+    }
+
+    fun clearPlayers() {
+        players.clear()
+        updatedSinceWorldSwitch = false
+        onKnownIsland = false
+    }
+
+    private fun shouldSendMsg(player: String): Boolean {
+        val cleanPlayer = player.removeColor()
+        return (
+            LorenzUtils.getPlayerName() != cleanPlayer &&
+                if (config.onlyKnownPeople) isPlayerKnown(cleanPlayer) ||
+                    (config.alwaysOnYourIsland && onPrivateIsland()) ||
+                    (config.alwaysOnKnownIslands && onKnownIsland)
+                else true
+            )
+    }
```

Comment:
```suggestion
        private fun shouldSendMsg(player: String): Boolean {
        val cleanPlayer = player.removeColor()
        
        if (LorenzUtils.getPlayerName() == cleanPlayer) return false
        if (!config.onlyKnownPeople) return true
        return isPlayerKnown(cleanPlayer) || (config.alwaysOnYourIsland && onPrivateIsland()) || (config.alwaysOnKnownIslands && onKnownIsland)
    }
```

Haven't tested this new function but this feels cleaner and more like existing code style